src/coreclr/jit/compiler.cpp

```diff
@@ -5524,24 +5524,163 @@ int Compiler::compCompile(CORINFO_MODULE_HANDLE classPtr,
 
     if (!info.compMatchedVM)
```

Comment:
CC. @davidwrighton; would be good if you could review this since it modifies the logic you originally added.

I'd expect this is never hit for Crossgen/Crossgen2/R2R or NativeAOT as they should all have matching VMs and so I'd expect this is largely just hit for `AltJit`, is that correct?

Is there a different or better check we could do to ensure this covers "any altJit" (such as x64 on x64 even) and still ensure we don't impact the scenarios we shouldn't?

Should we add a comment explicitly clarifying this consideration (that this shouldn't impact CG2/R2R/NAOT)?